lib/classroom_api.rb

```diff
@@ -8,190 +11,229 @@ def initialize(access_token)
   end
 
   def add_facility_id_to_classrooms
-    @rooms_in_db = Room.all.pluck(:rmrecnbr)
-    result = get_classrooms_list
-    if result['success']
-      classrooms_list = result['data'] 
-      number_of_api_calls_per_minutes = 0
-      redo_loop_number = 1
-      classrooms_list.each do |room|
-        # update only rooms for campuses and buildings from the MClassroom database
-        if @buildings_ids.include?(room['BuildingID'].to_i)
-          if number_of_api_calls_per_minutes < 400
-            number_of_api_calls_per_minutes += 1
-          else
-            @log.api_logger.debug "add_facility_id_to_classrooms, the script sleeps after #{number_of_api_calls_per_minutes} calls"
-            number_of_api_calls_per_minutes = 1
-            sleep(61.seconds)
-          end
-          facility_id = room['FacilityID'].to_s
-          # add facility_id and number of seats
-          result = get_classroom_info(ERB::Util.url_encode(facility_id))
-          if result['errorcode'] == "ERR429"
-            @log.api_logger.debug "add_facility_id_to_classrooms, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls"
-            number_of_api_calls_per_minutes = 0
-            sleep(61.seconds)
-            if redo_loop_number > 9
-              @debug = true
-              @log.api_logger.debug "add_facility_id_to_classrooms, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls #{redo_loop_number} times "
-              return @debug
+    begin
+      @rooms_in_db = Room.all.pluck(:rmrecnbr)
+      result = get_classrooms_list
+      if result['success']
+        classrooms_list = result['data']
+        number_of_api_calls_per_minutes = 0
+        redo_loop_number = 1
+        classrooms_list.each do |room|
+          # update only rooms for campuses and buildings from the MClassroom database
+          if @buildings_ids.include?(room['BuildingID'].to_i)
+            if number_of_api_calls_per_minutes < 400
```

Comment:
Consider creating constant variable for this value.

Reply:
done in [6158ce0](https://github.com/lsa-mis/mi_classrooms/pull/326/commits/6158ce0a5e3b305c2ca803b248e447d8701300f2)